File: HRMS/Models/DepartmentInsert.cs
﻿namespace HRMS.Models;

public class DepartmentInsert
{
    public string Name { get; set; } = string.Empty;

    public int? ParentDepartmentId { get; set; }

    public int Level { get; set; }

    public int? SupervisorId { get; set; }
}

File: HRMS.Desktop/lib/models/requests/department_insert.dart
import 'package:json_annotation/json_annotation.dart';

part 'department_insert.g.dart';

@JsonSerializable()
class DepartmentInsert {
  String name;
  int? parentDepartmentId;
  int? supervisorId;

  DepartmentInsert(
    this.name,
    this.parentDepartmentId,
    this.supervisorId,
  );

  factory DepartmentInsert.fromJson(Map<String, dynamic> json) =>
      _$DepartmentInsertFromJson(json);

  Map<String, dynamic> toJson() => _$DepartmentInsertToJson(this);
}
